src/Functions/getScalar.cpp

```diff
@@ -60,11 +61,89 @@ class FunctionGetScalar : public IFunction, WithContext
     mutable ColumnWithTypeAndName scalar;
 };
 
+
+/** Get special scalar values
+  */
+template <typename Scalar>
+class FunctionGetSpecialScalar : public IFunction, WithContext
```

Comment:
Looks like we don't actually need `WithContext` here.
Context is used only to get scalar in the constructor

Reply:
Indeed.